pybossa/importers/csv.py

```diff
@@ -64,26 +91,37 @@ def _convert_row_to_task_data(self, row, row_number):
                         task_data[header] = {}
                 else:
                     task_data[header] = cell
-            elif header.endswith('_priv_gold'):
-                if cell:
-                    field_name = re.sub('_priv_gold$', '', header)
-                    task_data.setdefault('gold_answers', {})[field_name] = cell
-            elif header.endswith('_gold'):
+                continue
+
+            gold_match = re.match('(?P<field>.*?)(_priv)?_gold(_(?P<type>json|number|bool|null))?$', header)
+            if gold_match:
                 if cell:
-                    field_name = re.sub('_gold$', '', header)
-                    task_data.setdefault('gold_answers', {})[field_name] = cell
-            elif header.endswith('_priv'):
+                    data_type = gold_match.group('type')
+                    field_name = gold_match.group('field')
+                    task_data.setdefault('gold_answers', {})[field_name] = get_value(header, cell, data_type)
+                continue
+
+            priv_match = re.match('(?P<field>.*?)_priv(_(?P<type>json|number|bool|null))?$', header)
+            if priv_match:
                 if cell:
-                    field_name = re.sub('_priv$', '', header)
-                    if data_access_levels:
-                        private_fields[field_name] = cell
+                    data_type = priv_match.group('type')
+                    field_name = priv_match.group('field')
+                    if data_access_levels: # This is how we check for private GIGwork.
+                        private_fields[field_name] = get_value(header, cell, data_type)
                     else:
-                        task_data["info"][field_name] = cell
-            elif header == 'data_access' and data_access_levels:
+                        task_data["info"][field_name] = get_value(header, cell, data_type)
```

Comment:
This path is not tested maybe will be good to add a test for it.

Reply:
Good point.